File: miscellaneous/sorting/selection-sort/20221224/selection_sort.py
from typing import List


def selection_sort(nums: List[int]) -> List[int]:
    if len(nums) <= 1:
        return nums

    for i in range(len(nums) - 1):
        smallest = i
        for j in range(i + 1, len(nums)):
            if nums[j] < nums[smallest]:
                smallest = j

        if smallest != i:
            [nums[i], nums[smallest]] = [nums[smallest], nums[i]]

    return nums

File: miscellaneous/sorting/selection-sort/20221224/selection-sort.js
'use strict';

module.exports = (nums) => {
    for (let i = 0; i < nums.length - 1; ++i) {
        let sorted = true;
        let smallest = i;

        for (let j = i + 1; j < nums.length; ++j) {
            if (nums[j] < nums[smallest]) {
                sorted = false;
                smallest = j;
            }
        }

        if (sorted) return nums;

        smallest !== i && ([nums[smallest], nums[i]] = [nums[i], nums[smallest]]);
    }

    return nums;
}